fraud/pb/proof.pb.go

```diff
@@ -124,31 +124,143 @@ func (m *BadEncoding) GetAxis() Axis {
 	return Axis_ROW
 }
 
+type ProofResponse struct {
+	Type  int32  `protobuf:"varint,1,opt,name=Type,proto3" json:"Type,omitempty"`
+	Value []byte `protobuf:"bytes,2,opt,name=Value,proto3" json:"Value,omitempty"`
+}
+
+func (m *ProofResponse) Reset()         { *m = ProofResponse{} }
+func (m *ProofResponse) String() string { return proto.CompactTextString(m) }
+func (*ProofResponse) ProtoMessage()    {}
+func (*ProofResponse) Descriptor() ([]byte, []int) {
+	return fileDescriptor_318cb87a8bb2d394, []int{1}
+}
+func (m *ProofResponse) XXX_Unmarshal(b []byte) error {
+	return m.Unmarshal(b)
+}
+func (m *ProofResponse) XXX_Marshal(b []byte, deterministic bool) ([]byte, error) {
+	if deterministic {
+		return xxx_messageInfo_ProofResponse.Marshal(b, m, deterministic)
+	} else {
+		b = b[:cap(b)]
+		n, err := m.MarshalToSizedBuffer(b)
+		if err != nil {
+			return nil, err
+		}
+		return b[:n], nil
+	}
+}
+func (m *ProofResponse) XXX_Merge(src proto.Message) {
+	xxx_messageInfo_ProofResponse.Merge(m, src)
+}
+func (m *ProofResponse) XXX_Size() int {
+	return m.Size()
+}
+func (m *ProofResponse) XXX_DiscardUnknown() {
+	xxx_messageInfo_ProofResponse.DiscardUnknown(m)
+}
+
+var xxx_messageInfo_ProofResponse proto.InternalMessageInfo
+
+func (m *ProofResponse) GetType() int32 {
+	if m != nil {
+		return m.Type
+	}
+	return 0
+}
+
+func (m *ProofResponse) GetValue() []byte {
+	if m != nil {
+		return m.Value
+	}
+	return nil
+}
+
+type FraudMessage struct {
+	RequestedProofType []int32          `protobuf:"varint,1,rep,packed,name=RequestedProofType,proto3" json:"RequestedProofType,omitempty"`
```

Comment:
We shouldn't do this, it's very strange to allow a request for more than one kind of proof and to have the response be an array of proof responses (that isn't sorted), etc.

Reply:
ok, then I'd suggest to make a enum inside proto with proof types.(can make in a separate PR)

Reply:
> it's very strange to allow a request for more than one kind of proof and to have the response be an array of proof responses (that isn't sorted), etc.

It was not and was part of the issues explaining the design.